libs/core/src/lib/breadcrumb/breadcrumb.component.ts

```diff
@@ -18,11 +26,102 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
+        if (window.innerWidth <= this.previousWindowInnerWidth) {
+            // and the breadcrumbs extend past the window
+            if (this.elementRef.nativeElement.getBoundingClientRect().right >= window.innerWidth) {
+                this.collapseBreadcrumbs();
+            }
+        } else if (this.collapsedBreadcrumbItems.length) { // if the screen is getting bigger and there are collapsed breadcrumbs
+            this.expandBreadcrumbs();
+        }
+        this.previousWindowInnerWidth = window.innerWidth;
+    }
+
+    /**
+     * @hidden
+     *
+     * This function moves breadcrumbs from the original QueryList of BreadCrumbItemDirectives into the array of
+     * collapsedBreadcrumbItems.
+     * */
+    collapseBreadcrumbs(): void {
+        let i = 0;
+        // move the breadcrumb items into a collapsed menu one by one, until the last one is inside the window
+        while (this.elementRef.nativeElement.getBoundingClientRect().right >= window.innerWidth && i < this.breadcrumbItems.length) {
+            const breadcrumbItem = this.breadcrumbItems.filter((item, index) => index === i)[0];
+            if (this.collapsedBreadcrumbItems.indexOf(breadcrumbItem) === -1) {
+                this.collapsedBreadcrumbItems.push(breadcrumbItem);
+            }
+            // hide the original breadcrumb items that have been moved in to the collapsed menu
+            breadcrumbItem.elementRef.nativeElement.style.display = 'none';
+            i++;
+        }
+    }
+
+    /**
+     * @hidden
+     *
+     * This function removes breadcrumb items from the collapsedBreadcrumbItems array and makes them visible again in the
+     * original QueryList of BreadcrumbItemDirectives.
+     * */
+    expandBreadcrumbs(): void {
+        let breakLoop = false;
+        let i = 0;
+        const originalCollapsedLength = this.collapsedBreadcrumbItems.length;
+        while (this.elementRef.nativeElement.getBoundingClientRect().right < window.innerWidth &&
+                !breakLoop && i < originalCollapsedLength) {
+            // get the most recently collapsed breadcrumb
+            const collapsedItemToPop = this.collapsedBreadcrumbItems[this.collapsedBreadcrumbItems.length - 1];
+            // find the corresponding breadcrumb from the original breadcrumb item QueryList
+            const breadcrumbToCheck = this.breadcrumbItems.filter((item) => item === collapsedItemToPop)[0];
+            /*
+              set display: 'inline-block' and visibility: 'hidden' - this way, the parent breadcrumb component's width will
+              contain the width of the breadcrumb item we might pop, without actually making the item visible to the user
+             */
+            breadcrumbToCheck.elementRef.nativeElement.style.display = 'inline-block';
+            breadcrumbToCheck.elementRef.nativeElement.style.visibility = 'hidden';
+            if (this.elementRef.nativeElement.getBoundingClientRect().right < window.innerWidth) {
```

Comment:
Could you add possibility to pass some container element?  to avoid things like that
![image](https://user-images.githubusercontent.com/26483208/74437047-c4e28900-4e67-11ea-9582-26b0583067d2.png)

So if there is element, check if `right` is more/less than container's `right`